subject.py

```diff
@@ -0,0 +1,178 @@
+    """
+
+
+@schema
+class Strain(dj.Manual):
+    # <class 'subjects.models.Strain'>
+    definition = """
+    strain_id:			int             # strain id
+    ---
+    descriptive_name:		varchar(255)	# descriptive name
+    description:    		varchar(255)	# description
+    """
+
+
+@schema
+class Sequence(dj.Manual):
+    # <class 'subjects.models.Sequence'>
+    definition = """
+    sequence_id:		int             # sequence id
+    ---
+    base_pairs:			varchar(255)	# base pairs
+    description:		varchar(255)	# description
+    informal_name:		varchar(255)	# informal name
+    """
+
+
+@schema
+class Allele(dj.Manual):
+    # <class 'subjects.models.Allele'>
+    definition = """
+    allele_id:			int             # allele id
+    ---
+    standard_name:		varchar(255)	# standard name
+    informal_name:		varchar(255)	# informal name
+    """
+
+
+@schema
+class Line(dj.Manual):
+    # <class 'subjects.models.Line'>
+    # TODO: autoname index: basically autogenerates a unique sequence
+    definition = """
+    -> Species
+    -> Strain
+    -> Sequence
+    line_id:				int             # line id
+    ---
+    name:				varchar(255)	# name
+    description:			varchar(255)	# description
+    target_phenotype:			varchar(255)	# target phenotype
+    auto_name:				varchar(255)	# auto name
+    is_active:				boolean		# is active
+    subject_autoname_index:		integer		# subject autoname index
+    breeding_pair_autoname_index:	integer		# breeding pair autoname index
+    litter_autoname_index:		integer		# litter autoname index
+    """
+
+
+@schema
+class Source(dj.Manual):
+    # <class 'subjects.models.Source'>
+    definition = """
+    source_id:				int             # source id
+    ---
+    name:				varchar(255)	# name
+    description:			varchar(255)	# description
+    """
+
+
+@schema
+class SubjectRequest(dj.Manual):
+    # <class 'subjects.models.SubjectRequest'>
+    definition = """
+    -> reference.User
+    -> Line
+    subject_request_id:			int     	# subject request id
+    ---
+    count:				integer		# count
+    date_time:				date		# date time
+    due_date:				date		# due date
+    description:			varchar(255)	# description
+    """
+
+
+@schema
+class Subject(dj.Manual):
+    # <class 'subjects.models.Subject'>
+    # todo:
+    # - where did responsible_user go? RETEST ADDED
+    # - -> SubjectRequest or track subjects as part table of subject request?
+    #      RETEST ADDED
+
+    definition = """
+    subject_id:			int                     # subject id
+    ---
+    nickname:			varchar(255)		# nickname
+    -> Line
+    sex:			enum("M", "F", "U")	# sex
+    birth_date:			date			# birth date
+    death_date:			date			# death date
+    wean_date:			date			# wean date
+    genotype_date:		date			# genotype date
+    (responsible_user)          -> reference.User
+    (request)                   -> SubjectRequest(subject_request_id)
+    -> Source
+    lamis_cage:			integer			# lamis cage
```

Comment:
I have no idea what `lamis` stands for but I'm guessing that this is referring to a specific cage that the mouse is placed in. Assuming that this guess is correct, then again this is something that can vary over the life course of mouse, and it's a bad idea to include this in the `Subject` table. If anything, we should make `Caging` table that has `cage_id` and `-> Subject` and most likely `timestamp`  as primary key. As long as they would make a new entry into this table every time a mouse is placed into a new cage, the table will allow them to keep track of which mouse were in which cage.

Reply:
done